crypto/bn/bn_ppc.c

```diff
@@ -40,7 +40,7 @@ int bn_mul_mont(BN_ULONG *rp, const BN_ULONG *ap, const BN_ULONG *bp,
      * no opportunity to figure it out...
      */
 
-#if defined(_ARCH_PPC64)
+#if defined(_ARCH_PPC64) && !defined(__ILP32__)
```

Comment:
There are other instances of the use of this macro which look suspect to me, e.g. see crypto/sha/sha512.c

Also bn_local.h does this correctly, but in a different way:

````
#  elif defined(_ARCH_PPC64) && defined(SIXTY_FOUR_BIT_LONG)
````
Should we be consistent?

Reply:
IMO that is a different condition. I am not sure about them being equivalent.

Reply:
As for the other instances, I am not sure about them, I suppose they are either working for the platform from #17087 correctly or not compiled in for other reasons.

Reply:
Ok. We can run with this as is - but I'm not entirely convinced there isn't another issue lurking somewhere.